### DISCUSSION TITLE
Internal Server Error 500 when trying to add a custom domain

### DISCUSSION THREAD

**Discussion by lxup** [appwrite/appwrite#4997] [Q&A] [answered]
Yo ! I'm trying to add a custom domain for my Appwriteproject but it's not working. My setup is just an Appwrite instance with Docker on a VPS (Debian 11). I'm not sure what other info I can give you as I'm just starting to learn Appwrite.

This can maybe help :
```
root@localhost:~# docker exec appwrite doctor
  __   ____  ____  _  _  ____  __  ____  ____     __  __
 / _\ (  _ \(  _ \/ )( \(  _ \(  )(_  _)(  __)   (  )/  \
/    \ ) __/ ) __/\ /\ / )   / )(   )(   ) _)  _  )((  O )
\_/\_/(__)  (__)  (_/\_)(__\_)(__) (__) (____)(_)(__)\__/

👩‍⚕️ Running Appwrite Doctor for version 1.2.0 ...

Checking for production best practices...
🔴 Hostname has no public suffix (localhost)
🔴 CNAME target has no public suffix (localhost)
🟢 Using a unique secret key for encryption
🟢 App environment is set for production
🟢 Abuse protection is enabled
🟢 Console access limits are enabled
🔴 HTTPS force option is disabled
🔴 Logging adapter is disabled

Checking connectivity...
Database............connected 👍
Queue...............connected 👍
Cache...............connected 👍
SMTP.............disconnected 👎
StatsD..............connected 👍
InfluxDB............connected 👍

Checking volumes...
🟢 Uploads Volume is readable
🟢 Uploads Volume is writeable
🟢 Cache Volume is readable
🟢 Cache Volume is writeable
🟢 Config Volume is readable
🟢 Config Volume is writeable
🟢 Certs Volume is readable
🟢 Certs Volume is writeable

Checking disk space usage...
🟢 Uploads Volume has 73.50GB free space (10.27% used)
🟢 Cache Volume has 73.50GB free space (10.27% used)
🟢 Config Volume has 73.50GB free space (10.27% used)
🟢 Certs Volume has 73.50GB free space (10.27% used)

You are running the latest version of Appwrite! 🥳
```

Thanks !

**Comment by stnguyen90** [comment, marked as answer]
This usually happens because the main Appwrite domain is not configured via the `_APP_DOMAIN` and `_APP_DOMAIN_TARGET` [environment variables](https://appwrite.io/docs/environment-variables#general). [Custom Domains](https://appwrite.io/docs/custom-domains) are additional project-specific domains usually created when a web app needs 1st party cookies.

For help generating the SSL certificate for your main domain, please look at the [Certificates docs](https://appwrite.io/docs/certificates). If you're still having trouble, please join us on [Discord](https://appwrite.io/discord) as it's much easier to troubleshoot synchronously.
